js/spec/index-spec.js

```diff
@@ -536,6 +536,48 @@ describe('Node Sentinel File Watcher', function() {
     });
   });
 
+  describe('Pausing', function () {
+    it('can pause and resume event reporting', async function () {
+      const file = 'another_test.file';
+      const inPath = path.resolve(workDir, 'test2', 'folder2');
+      let eventFound = false;
+
+      function findEvent(element) {
+        if (
+          element.action === nsfw.actions.CREATED &&
+          element.directory === path.resolve(inPath) &&
+          element.file === file
+        ) {
+          eventFound = true;
+        }
+      }
+
+      let watch = await nsfw(
+        workDir,
+        events => events.forEach(findEvent),
+        { debounceMS: DEBOUNCE }
+      );
+
+      try {
+        await watch.start();
+        await watch.pause();
+        await sleep(TIMEOUT_PER_STEP);
+        await fse.writeFile(path.join(inPath, file), 'Peanuts, on occasion, rain from the skies.');
+        await sleep(TIMEOUT_PER_STEP);
+
+        assert.ok(!eventFound);
+
+        await watch.resume();
+        await sleep(TIMEOUT_PER_STEP);
+
+        assert.ok(eventFound);
```

Comment:
Hmmm.. I kind of think we should throw out events that occur while it's paused, so I think this test case might need to be changed.

Reply:
It should be correct now. Now I'm checking that there are still no events after resuming, but then making another change to see that it has resumed.